### PR TITLE
Update Sourcery, SwiftLint, SwiftFormat versions

### PR DESCRIPTION
**Pull request by kmahar**
Updates our various dependencies and makes some minor changes to appease formatter/linter.

### CODE REVIEW

File: .swiftformat [hunk 1]
```diff
@@ -36,3 +36,9 @@
 
 # the ordering doesn't match linter's
 --disable specifiers
+
+# conflicts with our Swiftlint explicit ACL rule
+--disable extensionAccessControl
+
+# conflicts with Swiftlint opening_brace rule
+--disable wrapMultilineStatementBraces
```
Comment (kmahar) [inline .swiftformat:44]: SwiftFormat wanted us to start using a newline for opening braces following multiline conditionals, e.g.
```swift
if let a = x,
    let b = y {
    // ...
}
```

would become 
```swift
if let a = x,
    let b = y 
{
    ...
}
```

SwiftLint did not like this because it wants the opening brace to be on the same line as the end of the statement. I have like a 3/10 preference for the way we've always done it so stuck with that, but if you feel strongly the other way happy to switch

File: Sources/MongoSwift/BSON/BSONDocument.swift [hunk 2]
```diff
@@ -255,7 +255,7 @@ extension BSONDocument {
             throw BSONError.InternalError(message: "No keys to exclude, use 'bson_copy' instead")
         }
 
-        let cStrings: [ContiguousArray<CChar>] = keys.map { $0.utf8CString }
+        let cStrings: [ContiguousArray<CChar>] = keys.map(\.utf8CString)
```
Comment (kmahar) [inline Sources/MongoSwift/BSON/BSONDocument.swift:258]: new SwiftFormat rule that prefers using keypaths where possible -- I like it so I left it on for now, LMK what you think

File: Tests/MongoSwiftSyncTests/CommandMonitoringTests.swift [hunk 3]
```diff
@@ -102,12 +102,12 @@ private struct CMTest: Decodable {
     // `context` dictionary so we can access it in future events for the same test
     var context = [String: Any]()
 
-    var expectations: [ExpectationType] { try! expectationDocs.map { try makeExpectation($0) } }
+    var expectations: [ExpectationType] { try! self.expectationDocs.map { try makeExpectation($0) } }
```
Comment (kmahar) [inline Tests/MongoSwiftSyncTests/CommandMonitoringTests.swift:105]: explicit self checking seems to have improved

File: Tests/MongoSwiftSyncTests/SpecTestRunner/SpecTest.swift [hunk 4]
```diff
@@ -1,3 +1,4 @@
+// swiftlint:disable duplicate_imports
```
Comment (kmahar) [inline Tests/MongoSwiftSyncTests/SpecTestRunner/SpecTest.swift:1]: I think this is because it doesn't distinguish between these two lines, even though they have different meanings:
```
import MongoSwiftSync
@testable import struct MongoSwiftSync.MongoClientOptions
```

**Comment by kmahar** [comment]
Evg failures should be fixed by #540

File: .swiftformat (hunk 1 above)
Comment (patrickfreed) [reply .swiftformat:44]: I prefer the latter (more than 3/10 :smile: ) because otherwise it's kinda hard to parse the difference between the `if` portion of the statement and the actual body (e.g. here https://github.com/mongodb/mongo-swift-driver/blob/636cd541225754a8956e42419461a433bd23b113/Sources/MongoSwift/MongoError.swift#L354).

Ideally the various `let` keywords would be aligned too, but I'm not sure if that's an option.

Comment (kmahar) [reply .swiftformat]: that's a good point. let's turn it on then 🙂

**Review by patrickfreed — APPROVED** [review]
